addons/web/static/lib/owl/owl.js

```diff
@@ -2870,7 +2870,9 @@
     function validateBaseType(key, value, type) {
         if (typeof type === "function") {
             if (typeof value === "object") {
-                if (!(value instanceof type)) {
+                const ownerDocument = (type === HTMLElement) && value.ownerDocument;
+                const htmlElement = ownerDocument && ownerDocument.defaultView.HTMLElement;
+                if (!(value instanceof type || htmlElement && value instanceof htmlElement)) {

```

Comment:
I think those three lines are quite confusing. I suggest:
```js
// Validate against the correct HTMLElement class with respect to iframes.
if (type === HTMLElement && value.ownerDocument && value.ownerDocument.defaultView) {
    type = value.ownerDocument.defaultView.HTMLElement
}
```

Though looking at previous blames inside `owl.js`, it seems @ged-odoo might want to commit that in their own repository ?

Reply:
@dmo-odoo I have updated the requested changes.

Reply:
@aab-odoo I suppose you'd rather do those changes in your own repository rather than us changing the `owl.js` file in Odoo, right ?

Reply:
Indeed... @sdegueldre

Reply:
I don't think Owl's behaviour is unexpected here, props validation shouldn't be doing any magic remapping between class instances of different frames. If a component is expected to work cross-frame, its props validation should be adapted to be more lax or use the correct frame's HTMLElement class.

For this particular issue, the proper fix is probably to backport https://github.com/odoo/odoo/pull/116499

Reply:
Since this is more of a framework issue regardless of how you choose to support that kind of things with iframes, I'll let you handle it the way you prefer.